dotnetv3/Support/Scenarios/SupportCaseScenario.cs

```diff
@@ -0,0 +1,391 @@
+
+        return service.Categories[choiceNumber - 1];
+    }
+
+    /// <summary>
+    /// List available severity levels from AWS support, and select a level for the example.
+    /// </summary>
+    /// <returns>The selected severity level.</returns>
+    private static async Task<SeverityLevel> DisplayAndSelectSeverity()
+    {
+        Console.WriteLine(new string('-', 80));
+        var severityLevels = await _supportWrapper.DescribeSeverityLevels();
+
+        Console.WriteLine($"3. Get and display available severity levels:");
+        for (int i = 0; i < 10 && i < severityLevels.Count; i++)
+        {
+            Console.WriteLine($"\t{i + 1}. {severityLevels[i].Name}");
+        }
+
+        var choiceNumber = 0;
+        while (choiceNumber < 1 || choiceNumber > severityLevels.Count)
+        {
+            Console.WriteLine(
+                "Select an example severity level by entering a number from the list above:");
+            var choice = Console.ReadLine();
+            Int32.TryParse(choice, out choiceNumber);
+        }
+        Console.WriteLine(new string('-', 80));
+
+        return severityLevels[choiceNumber - 1];
+    }
+
+    /// <summary>
+    /// Create an example support case.
+    /// </summary>
+    /// <param name="service">Service to use for the new case.</param>
+    /// <param name="category">Category to use for the new case.</param>
+    /// <param name="severity">Severity to use for the new case.</param>
+    /// <returns>The caseId of the new support case.</returns>
+    private static async Task<string> CreateSupportCase(Service service,
+        Category category, SeverityLevel severity)
+    {
+        Console.WriteLine(new string('-', 80));
+        Console.WriteLine($"4. Create an example support case" +
+                          $" with the following settings:" +
+                          $" \n\tService: {service.Name}, Category: {category.Name} " +
+                          $"and Severity Level: {severity.Name}.");
+        var caseId = await _supportWrapper.CreateCase(service.Code, category.Code, severity.Code,
+            "Example case for testing, please ignore.", "This is my example support case.");
+
+        Console.WriteLine($"\tNew case created with id {caseId}");
+
+        Console.WriteLine(new string('-', 80));
+
+        return caseId;
+    }
+
+    /// <summary>
+    /// List open cases for the current day.
+    /// </summary>
+    /// <returns>Async task.</returns>
+    private static async Task DescribeTodayOpenCases()
+    {
+        Console.WriteLine($"5. List the open support cases for the current day.");
+        // Describe the cases. In case it is empty, try again and allow time for the new case to appear.
+        List<CaseDetails> currentOpenCases = null!;
+        while (currentOpenCases == null || currentOpenCases.Count == 0)
+        {
+            Thread.Sleep(1000);
+            currentOpenCases = await _supportWrapper.DescribeCases(
```

Comment:
This call might be paginated. Probably not necessary to show that in this demo? Unless there's a risk of lots of cases being returned, for some reason, and messing up the demo.

Reply:
As we discussed for this one, the use of the paginator is inside the wrapper.